part2/phonebook/src/App.js

```diff
@@ -0,0 +1,172 @@
+import React, { useState, useEffect } from 'react'
+import Persons from './components/Persons'
+import Filter from './components/Filter'
+import PersonForm from './components/PersonForm'
+import Notification from './components/Notification'
+import Error from './components/Error'
+import personService from './services/persons'
+
+
+
+
+const App = () => {
+    const bodyStyle = {
+        fontFamily: `"Lucida Sans", "Lucida Sans Regular", "Lucida Grande",
+            "Lucida Sans Unicode", Geneva, Verdana, sansSerif`
+    }
+
+    const h1Style = {
+        // jus! hex aiziet
+        color: `#0c1616`,
+        textTransform: 'uppercase',
+        textAlign: 'center'
+    }
+
+    const h2Style = {
+        color: `#021818`,
+        textTransform: 'uppercase'
+    }
+
+    const [persons, setPersons] = useState([])
+    const [newName, setNewName] = useState('')
+    const [newNumber, setNewNumber] = useState('')
+    const [searchName, setSearchName] = useState('')
+    const [showAll, setShowAll] = useState(true)
+    const [message, setMessage] = useState(null)
+    const [error, setErrorMessage] = useState(null)
+
+
+    useEffect(() => {
+        personService
+            .getAll()
+            // https://javascript.info/promise-chaining
+            .then(initialData => {
+                setPersons(initialData)
+            })
+    }, [])
+
+
+
+
+    const addContact = (e) => {
+        e.preventDefault()
+
+        const newObject = {
+            name: newName,
+            number: newNumber
+        }
+
+
+        if (persons.some(person => person.name === newName)) {
+            const person = persons.find(p => p.name === newName)
+            const updatedPerson = { ...person, number: newNumber }
+
+            window.confirm(`${newName} is already in phonebook. Replace numbers?`)
+
+            personService
+                .update(updatedPerson.id, updatedPerson)
+                .then(
+                    setPersons(persons.map(p => p.id !== updatedPerson.id ? p : updatedPerson)),
+                    setNewName(''),
+                    setNewNumber(''),
+                    setMessage(`${newName}'s number is replaced`),
+                    setTimeout(() => {
+                        setMessage(null)
+                    }, 5000))
```

Comment:
Šis galīgi nav pareizi.
Tam vajadzētu būt `then` callback'ā, nevis padotiem kā argumentiem uz `then`. ;)

Šķiet, ka pat neesi notestējusi, jo būtu pamanījusi, ka kaut kas shady is going on.

Reply:
Nē, nē, es testēju visu, tiešām! Ha. Jā, man vairāk jāiedziļinās promise chain. Atgriezīšos pie tā.

Reply:
Tie komati man tolaik palīdzēja izslīdēt cauri